src/tests/ftest/util/server_utils.py

```diff
@@ -633,10 +634,11 @@ def check_rank_state(self, rank, valid_state, max_checks=1):
                 # The regex failed to get the rank and state
                 raise ServerFailed("Error obtaining {} output: {}".format(self.dmg, data))
             checks += 1
-            if data[rank]["state"] == valid_state:
-                return True
+            for rank in ranks:
+                if data[rank]["state"] == valid_state:
+                    return [True]
 
-        return False
+        return [False, rank]
```

Comment:
Do you mean to return only the first failed rank? Why not all failed ranks? Also, could we support multiple states? Something like:

```
while checks < max_checks:
    ...
    failed_ranks = []
    for rank in ranks:
        if data[rank]["state"] not in valid_states:
            failed_ranks.append(rank)
    if not failed_ranks:
        return []
```

Reply:
Sure, recommended changes have been incorporated. Please review the latest push.